specs/searchable-asn-group/searchable-asn-group.md

```diff
@@ -0,0 +1,112 @@
+# Searchable ASN Filters (ff-asn-group)
+
+Some data sets have huge amounts of filters per filter group (150+). Too many to display all at once and too many to browse by eye.
+
+To improve usability the `ff-asn-group` gets an _optional_ search field. It is scoped to its associated filter group.
+
+
+## General
+
+This feature is an addition to the existing `ff-asn-group`. HTML examples are limited to new parts and minimal context for focus.
+
+The new search field is always the first element inside the collapsible `.ffw-wrapper` container. This means the filter group must be expanded in order to use it.
+
+Users may specify a custom search field template. It is defined via the `slot="filterSearch"` attribute. As it is a slot, its location inside the `ff-asn-group` template is irrelevant.
+
+## Features
+
+#### search as you type
+
+Filter items are searched for while user is typing in search field. While search term is present, filter group is kind of in _"search mode"_ compared to _"regular mode"_.
```

Comment:
Will the search be triggered with the first key stroke? Maybe the result list is also too long in this case. Consider to introduce a configurable threshold value for this setting.

Reply:
I was indeed thinking about starting to search from the first key stroke and leave size limitation to CSS.
Not every first character would yield the same list size. While `"a"` might produce 200 results which is not helpful, `"q"` might produce only 15 where instant feedback would certainly be appreciated.
Although 200 results even in a scrollable list is still large, users would get instant feedback about their query which has a certain value.

Do you think it's still more valuable to have another attribute to configure this?
Implementing it shouldn't be a big deal. I'm rather concerned about overwhelming WebComponents users with configuration options. We already have quite a few 🙂